[Automation] java run fails for org.clojure:google-closure-library-third-party:0.0-1376

## java run on ubuntu-latest / JDK 25 / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: 25
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=0.0-1376 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean javaTest -Pcoordinates=org.clojure:google-closure-library-third-party:0.0-20250515-f04e4c0e`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/28121378337
Last 300 lines of the log:
```console
     * @return {!goog.async.Deferred} The deferred object for chaining.
     */
    goog.async.Deferred.prototype.awaitDeferred = function(otherDeferred) {
      return this.addCallback(goog.bind(otherDeferred.branch, otherDeferred));
    };


    /**
     * Create a branch off this Deferred's callback chain, and return it as a new
     * Deferred. This means that the return value will have the value at the current
     * point in the callback chain, regardless of any further callbacks added to
     * this Deferred.
     *
     * Additional callbacks added to the original Deferred will not affect the value
     * of any branches. All branches at the same stage in the callback chain will
     * receive the same starting value.
     *
     * @param {boolean=} opt_propagateCancel If cancel() is called on every child
     *     branch created with opt_propagateCancel, the parent will be cancelled as
     *     well.
     * @return {!goog.async.Deferred} The deferred value at this point in the
     *     callback chain.
     */
    goog.async.Deferred.prototype.branch = function(opt_propagateCancel) {
      var d = new goog.async.Deferred();
      this.chainDeferred(d);
      if (opt_propagateCancel) {
        d.parent_ = this;
        this.branches_++;
      }
      return d;
    };


    /**
     * Registers a function as both callback and errback.
     * @param {!Function} f The function to be called on any result.
     * @param {Object=} opt_scope An optional scope to call the callbacks in.
     * @return {!goog.async.Deferred} The deferred object for chaining.
     */
    goog.async.Deferred.prototype.addBoth = function(f, opt_scope) {
      return this.addCallbacks(f, f, opt_scope);
    };


    /**
     * @return {boolean} Whether callback or errback has been called on this
     *     deferred.
     */
    goog.async.Deferred.prototype.hasFired = function() {
      return this.fired_;
    };


    /**
     * @param {*} res The current callback result.
     * @return {boolean} Whether the current result is an error that should cause
     *     registered errbacks to fire. May be overridden by subclasses to handle
     *     special error types.
     * @protected
     */
    goog.async.Deferred.prototype.isError = function(res) {
      return res instanceof Error;
    };


    /**
     * @return {boolean} Whether an errback has been registered.
     * @private
     */
    goog.async.Deferred.prototype.hasErrback_ = function() {
      return goog.array.some(this.chain_, function(chainRow) {
        // The errback is the second element in the array.
        return goog.isFunction(chainRow[1]);
      });
    };


    /**
     * Exhausts the callback sequence once a result is available.
     * @private
     */
    goog.async.Deferred.prototype.fire_ = function() {
      if (this.unhandledExceptionTimeoutId_ && this.hasFired() &&
          this.hasErrback_()) {
        // It is possible to add errbacks after the Deferred has fired. If a new
        // errback is added immediately after the Deferred encountered an unhandled
        // error, but before that error is rethrown, cancel the rethrow.
        goog.global.clearTimeout(this.unhandledExceptionTimeoutId_);
        delete this.unhandledExceptionTimeoutId_;
      }

      if (this.parent_) {
        this.parent_.branches_--;
        delete this.parent_;
      }

      var res = this.result_;
      var unhandledException = false;
      var isChained = false;

      while (this.chain_.length && this.paused_ == 0) {
        var chainEntry = this.chain_.shift();

        var callback = chainEntry[0];
        var errback = chainEntry[1];
        var scope = chainEntry[2];

        var f = this.hadError_ ? errback : callback;
        if (f) {
          try {
            var ret = f.call(scope || this.defaultScope_, res);

            // If no result, then use previous result.
            if (goog.isDef(ret)) {
              // Bubble up the error as long as the return value hasn't changed.
              this.hadError_ = this.hadError_ && (ret == res || this.isError(ret));
              this.result_ = res = ret;
            }

            if (res instanceof goog.async.Deferred) {
              isChained = true;
              this.pause_();
            }

          } catch (ex) {
            res = ex;
            this.hadError_ = true;

            if (!this.hasErrback_()) {
              // If an error is thrown with no additional errbacks in the queue,
              // prepare to rethrow the error.
              unhandledException = true;
            }
          }
        }
      }

      this.result_ = res;

      if (isChained && this.paused_) {
        res.addCallbacks(
            goog.bind(this.continue_, this, true /* isSuccess */),
            goog.bind(this.continue_, this, false /* isSuccess */));
        res.chained_ = true;
      }

      if (unhandledException) {
        // Rethrow the unhandled error after a timeout. Execution will continue, but
        // the error will be seen by global handlers and the user. The rethrow will
        // be canceled if another errback is appended before the timeout executes.
        this.unhandledExceptionTimeoutId_ = goog.global.setTimeout(function() {
          // The stack trace is clobbered when the error is rethrown. Append the
          // stack trace to the message if available. Since no one is capturing this
          // error, the stack trace will be printed to the debug console.
          if (goog.DEBUG && goog.isDef(res.message) && res.stack) {
            res.message += '\n' + res.stack;
          }
          throw res;
        }, 0);
      }
    };


8 tests completed, 6 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.clojure/google-closure-library-third-party/0.0-20250515-f04e4c0e/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s

    /**
     * Creates a deferred that always succeeds.
     * @param {*} res The result.
     * @return {!goog.async.Deferred} The deferred object.
     */
    goog.async.Deferred.succeed = function(res) {
      var d = new goog.async.Deferred();
      d.callback(res);
      return d;
    };


    /**
     * Creates a deferred that always fails.
     * @param {*} res The error result.
     * @return {!goog.async.Deferred} The deferred object.
     */
    goog.async.Deferred.fail = function(res) {
      var d = new goog.async.Deferred();
      d.errback(res);
      return d;
    };


    /**
     * An error sub class that is used when a deferred has already been called.
     * @param {!goog.async.Deferred} deferred The deferred object.
     * @constructor
     * @extends {goog.debug.Error}
     */
    goog.async.Deferred.AlreadyCalledError = function(deferred) {
      goog.debug.Error.call(this);

      /**
       * The deferred that raised this error.
       * @type {goog.async.Deferred}
       */
      this.deferred = deferred;
    };
    goog.inherits(goog.async.Deferred.AlreadyCalledError, goog.debug.Error);


    /**
     * Message text.
     * @type {string}
     * @override
     */
    goog.async.Deferred.AlreadyCalledError.prototype.message = 'Already called';



    /**
     * An error sub class that is used when a deferred is cancelled.
     * @param {!goog.async.Deferred} deferred The deferred object.
     * @constructor
     * @extends {goog.debug.Error}
     */
    goog.async.Deferred.CancelledError = function(deferred) {
      goog.debug.Error.call(this);

      /**
       * The deferred that raised this error.
       * @type {goog.async.Deferred}
       */
      this.deferred = deferred;
    };
    goog.inherits(goog.async.Deferred.CancelledError, goog.debug.Error);


    /**
     * Message text.
     * @type {string}
     * @override
     */
    goog.async.Deferred.CancelledError.prototype.message = 'Deferred was cancelled';
    "
    to contain:
      "goog.require('goog.Promise')" 
        at org_clojure.google_closure_library_third_party.Google_closure_library_third_partyTest.closureModulesKeepExpectedThirdPartyIntegrationPoints(Google_closure_library_third_partyTest.java:76)

Google_closure_library_third_partyTest > distributedResourceSetIsAvailableOnTheClasspath() FAILED
    java.lang.AssertionError: [resource goog/dojo/dom/query_test_dom.html should exist] 
    Expecting actual not to be null
        at org_clojure.google_closure_library_third_party.Google_closure_library_third_partyTest.readResource(Google_closure_library_third_partyTest.java:198)
        at org_clojure.google_closure_library_third_party.Google_closure_library_third_partyTest.distributedResourceSetIsAvailableOnTheClasspath(Google_closure_library_third_partyTest.java:52)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.clojure/google-closure-library-third-party/0.0-20250515-f04e4c0e/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 3 executed, 10 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 0.0-1376 at 2026-06-24T18:49:49Z with exit code 1 after 4s
❌ ERROR: 'java run' for 0.0-1376 failed with exit code 1.
FAILED[java run][0.0-1376][env GVM_TCK_LV=0.0-1376 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean javaTest -Pcoordinates=org.clojure:google-closure-library-third-party:0.0-20250515-f04e4c0e]
```

## java run on ubuntu-latest / JDK latest-ea / mode current-defaults

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: current-defaults
Reproducer: `env GVM_TCK_LV=0.0-1376 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean javaTest -Pcoordinates=org.clojure:google-closure-library-third-party:0.0-20250515-f04e4c0e`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/28121378337
Last 300 lines of the log:
```console
     * @return {!goog.async.Deferred} The deferred object for chaining.
     */
    goog.async.Deferred.prototype.awaitDeferred = function(otherDeferred) {
      return this.addCallback(goog.bind(otherDeferred.branch, otherDeferred));
    };


    /**
     * Create a branch off this Deferred's callback chain, and return it as a new
     * Deferred. This means that the return value will have the value at the current
     * point in the callback chain, regardless of any further callbacks added to
     * this Deferred.
     *
     * Additional callbacks added to the original Deferred will not affect the value
     * of any branches. All branches at the same stage in the callback chain will
     * receive the same starting value.
     *
     * @param {boolean=} opt_propagateCancel If cancel() is called on every child
     *     branch created with opt_propagateCancel, the parent will be cancelled as
     *     well.
     * @return {!goog.async.Deferred} The deferred value at this point in the
     *     callback chain.
     */
    goog.async.Deferred.prototype.branch = function(opt_propagateCancel) {
      var d = new goog.async.Deferred();
      this.chainDeferred(d);
      if (opt_propagateCancel) {
        d.parent_ = this;
        this.branches_++;
      }
      return d;
    };


    /**
     * Registers a function as both callback and errback.
     * @param {!Function} f The function to be called on any result.
     * @param {Object=} opt_scope An optional scope to call the callbacks in.
     * @return {!goog.async.Deferred} The deferred object for chaining.
     */
    goog.async.Deferred.prototype.addBoth = function(f, opt_scope) {
      return this.addCallbacks(f, f, opt_scope);
    };


    /**
     * @return {boolean} Whether callback or errback has been called on this
     *     deferred.
     */
    goog.async.Deferred.prototype.hasFired = function() {
      return this.fired_;
    };


    /**
     * @param {*} res The current callback result.
     * @return {boolean} Whether the current result is an error that should cause
     *     registered errbacks to fire. May be overridden by subclasses to handle
     *     special error types.
     * @protected
     */
    goog.async.Deferred.prototype.isError = function(res) {
      return res instanceof Error;
    };


    /**
     * @return {boolean} Whether an errback has been registered.
     * @private
     */
    goog.async.Deferred.prototype.hasErrback_ = function() {
      return goog.array.some(this.chain_, function(chainRow) {
        // The errback is the second element in the array.
        return goog.isFunction(chainRow[1]);
      });
    };


    /**
     * Exhausts the callback sequence once a result is available.
     * @private
     */
    goog.async.Deferred.prototype.fire_ = function() {
      if (this.unhandledExceptionTimeoutId_ && this.hasFired() &&
          this.hasErrback_()) {
        // It is possible to add errbacks after the Deferred has fired. If a new
        // errback is added immediately after the Deferred encountered an unhandled
        // error, but before that error is rethrown, cancel the rethrow.
        goog.global.clearTimeout(this.unhandledExceptionTimeoutId_);
        delete this.unhandledExceptionTimeoutId_;
      }

      if (this.parent_) {
        this.parent_.branches_--;
        delete this.parent_;
      }

      var res = this.result_;
      var unhandledException = false;
      var isChained = false;

      while (this.chain_.length && this.paused_ == 0) {
        var chainEntry = this.chain_.shift();

        var callback = chainEntry[0];
        var errback = chainEntry[1];
        var scope = chainEntry[2];

        var f = this.hadError_ ? errback : callback;
        if (f) {
          try {
            var ret = f.call(scope || this.defaultScope_, res);

            // If no result, then use previous result.
            if (goog.isDef(ret)) {
              // Bubble up the error as long as the return value hasn't changed.
              this.hadError_ = this.hadError_ && (ret == res || this.isError(ret));
              this.result_ = res = ret;
            }

            if (res instanceof goog.async.Deferred) {
              isChained = true;
              this.pause_();
            }

          } catch (ex) {
            res = ex;
            this.hadError_ = true;

            if (!this.hasErrback_()) {
              // If an error is thrown with no additional errbacks in the queue,
              // prepare to rethrow the error.
              unhandledException = true;
            }
          }
        }
      }

      this.result_ = res;

      if (isChained && this.paused_) {
        res.addCallbacks(
            goog.bind(this.continue_, this, true /* isSuccess */),
            goog.bind(this.continue_, this, false /* isSuccess */));
        res.chained_ = true;
      }

      if (unhandledException) {
        // Rethrow the unhandled error after a timeout. Execution will continue, but
        // the error will be seen by global handlers and the user. The rethrow will
        // be canceled if another errback is appended before the timeout executes.
        this.unhandledExceptionTimeoutId_ = goog.global.setTimeout(function() {
          // The stack trace is clobbered when the error is rethrown. Append the
          // stack trace to the message if available. Since no one is capturing this
          // error, the stack trace will be printed to the debug console.
          if (goog.DEBUG && goog.isDef(res.message) && res.stack) {
            res.message += '\n' + res.stack;
          }
          throw res;
        }, 0);
      }
    };


    /**
     * Creates a deferred that always succeeds.
     * @param {*} res The result.
     * @return {!goog.async.Deferred} The deferred object.
     */
    goog.async.Deferred.succeed = function(res) {
      var d = new goog.async.Deferred();
      d.callback(res);
      return d;
    };


    /**
     * Creates a deferred that always fails.
     * @param {*} res The error result.
     * @return {!goog.async.Deferred} The deferred object.
     */
    goog.async.Deferred.fail = function(res) {
      var d = new goog.async.Deferred();
      d.errback(res);
      return d;
    };


    /**
     * An error sub class that is used when a deferred has already been called.
     * @param {!goog.async.Deferred} deferred The deferred object.
     * @constructor
     * @extends {goog.debug.Error}
     */
    goog.async.Deferred.AlreadyCalledError = function(deferred) {
      goog.debug.Error.call(this);

      /**
       * The deferred that raised this error.
       * @type {goog.async.Deferred}
       */
      this.deferred = deferred;
    };
    goog.inherits(goog.async.Deferred.AlreadyCalledError, goog.debug.Error);


    /**
     * Message text.
     * @type {string}
     * @override
     */
    goog.async.Deferred.AlreadyCalledError.prototype.message = 'Already called';



    /**
     * An error sub class that is used when a deferred is cancelled.
     * @param {!goog.async.Deferred} deferred The deferred object.
     * @constructor
     * @extends {goog.debug.Error}
     */
    goog.async.Deferred.CancelledError = function(deferred) {
      goog.debug.Error.call(this);

      /**
       * The deferred that raised this error.
       * @type {goog.async.Deferred}
       */
      this.deferred = deferred;
    };
    goog.inherits(goog.async.Deferred.CancelledError, goog.debug.Error);


    /**
     * Message text.
     * @type {string}
     * @override
     */
    goog.async.Deferred.CancelledError.prototype.message = 'Deferred was cancelled';
    "
    to contain:
      "goog.require('goog.Promise')" 
        at org_clojure.google_closure_library_third_party.Google_closure_library_third_partyTest.closureModulesKeepExpectedThirdPartyIntegrationPoints(Google_closure_library_third_partyTest.java:76)

Google_closure_library_third_partyTest > distributedResourceSetIsAvailableOnTheClasspath() FAILED
    java.lang.AssertionError: [resource goog/dojo/dom/query_test_dom.html should exist] 
    Expecting actual not to be null
        at org_clojure.google_closure_library_third_party.Google_closure_library_third_partyTest.readResource(Google_closure_library_third_partyTest.java:198)
        at org_clojure.google_closure_library_third_party.Google_closure_library_third_partyTest.distributedResourceSetIsAvailableOnTheClasspath(Google_closure_library_third_partyTest.java:52)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.clojure/google-closure-library-third-party/0.0-20250515-f04e4c0e/build/reports/problems/problems-report.html

8 tests completed, 6 failed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.clojure/google-closure-library-third-party/0.0-20250515-f04e4c0e/build/reports/tests/test/index.html

* Try:
> Run with --scan to generate a Build Scan (Powered by Develocity).

BUILD FAILED in 2s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 3 executed, 10 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 0.0-1376 at 2026-06-24T18:49:52Z with exit code 1 after 5s
❌ ERROR: 'java run' for 0.0-1376 failed with exit code 1.
FAILED[java run][0.0-1376][env GVM_TCK_LV=0.0-1376 GVM_TCK_NATIVE_IMAGE_MODE=current-defaults ./gradlew clean javaTest -Pcoordinates=org.clojure:google-closure-library-third-party:0.0-20250515-f04e4c0e]
```

## java run on ubuntu-latest / JDK latest-ea / mode future-defaults-all

Failure kind: java run
OS: ubuntu-latest
JDK: latest-ea
Native-image mode: future-defaults-all
Reproducer: `env GVM_TCK_LV=0.0-1376 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean javaTest -Pcoordinates=org.clojure:google-closure-library-third-party:0.0-20250515-f04e4c0e`
Runner log: https://github.com/oracle/graalvm-reachability-metadata/actions/runs/28121378337
Last 300 lines of the log:
```console
          otherDeferred.callback, otherDeferred.errback, otherDeferred);
      return this;
    };


    /**
     * Makes this Deferred wait for otherDeferred to be called, and its preceding
     * callbacks to be executed, before continuing with the callback sequence.
     *
     * This is equivalent to adding a callback that returns otherDeferred, but
     * doesn't prevent additional callbacks from being added to otherDeferred.
     *
     * @param {!goog.async.Deferred} otherDeferred The Deferred to wait for.
     * @return {!goog.async.Deferred} The deferred object for chaining.
     */
    goog.async.Deferred.prototype.awaitDeferred = function(otherDeferred) {
      return this.addCallback(goog.bind(otherDeferred.branch, otherDeferred));
    };


    /**
     * Create a branch off this Deferred's callback chain, and return it as a new
     * Deferred. This means that the return value will have the value at the current
     * point in the callback chain, regardless of any further callbacks added to
     * this Deferred.
     *
     * Additional callbacks added to the original Deferred will not affect the value
     * of any branches. All branches at the same stage in the callback chain will
     * receive the same starting value.
     *
     * @param {boolean=} opt_propagateCancel If cancel() is called on every child
     *     branch created with opt_propagateCancel, the parent will be cancelled as
     *     well.
     * @return {!goog.async.Deferred} The deferred value at this point in the
     *     callback chain.
     */
    goog.async.Deferred.prototype.branch = function(opt_propagateCancel) {
      var d = new goog.async.Deferred();
      this.chainDeferred(d);
      if (opt_propagateCancel) {
        d.parent_ = this;
        this.branches_++;
      }
      return d;
    };


    /**
     * Registers a function as both callback and errback.
     * @param {!Function} f The function to be called on any result.
     * @param {Object=} opt_scope An optional scope to call the callbacks in.
     * @return {!goog.async.Deferred} The deferred object for chaining.
     */
    goog.async.Deferred.prototype.addBoth = function(f, opt_scope) {
      return this.addCallbacks(f, f, opt_scope);
    };


    /**
     * @return {boolean} Whether callback or errback has been called on this
     *     deferred.
     */
    goog.async.Deferred.prototype.hasFired = function() {
      return this.fired_;
    };


    /**
     * @param {*} res The current callback result.
     * @return {boolean} Whether the current result is an error that should cause
     *     registered errbacks to fire. May be overridden by subclasses to handle
     *     special error types.
     * @protected
     */
    goog.async.Deferred.prototype.isError = function(res) {
      return res instanceof Error;
    };


    /**
     * @return {boolean} Whether an errback has been registered.
     * @private
     */
    goog.async.Deferred.prototype.hasErrback_ = function() {
      return goog.array.some(this.chain_, function(chainRow) {
        // The errback is the second element in the array.
        return goog.isFunction(chainRow[1]);
      });
    };


    /**
     * Exhausts the callback sequence once a result is available.
     * @private
     */
    goog.async.Deferred.prototype.fire_ = function() {
      if (this.unhandledExceptionTimeoutId_ && this.hasFired() &&
          this.hasErrback_()) {
        // It is possible to add errbacks after the Deferred has fired. If a new
        // errback is added immediately after the Deferred encountered an unhandled
        // error, but before that error is rethrown, cancel the rethrow.
        goog.global.clearTimeout(this.unhandledExceptionTimeoutId_);
        delete this.unhandledExceptionTimeoutId_;
      }

      if (this.parent_) {
        this.parent_.branches_--;
        delete this.parent_;
      }

      var res = this.result_;
      var unhandledException = false;
      var isChained = false;

      while (this.chain_.length && this.paused_ == 0) {
        var chainEntry = this.chain_.shift();

        var callback = chainEntry[0];
        var errback = chainEntry[1];
        var scope = chainEntry[2];

        var f = this.hadError_ ? errback : callback;
        if (f) {
          try {
            var ret = f.call(scope || this.defaultScope_, res);

            // If no result, then use previous result.
            if (goog.isDef(ret)) {
              // Bubble up the error as long as the return value hasn't changed.
              this.hadError_ = this.hadError_ && (ret == res || this.isError(ret));
              this.result_ = res = ret;
            }

            if (res instanceof goog.async.Deferred) {
              isChained = true;
              this.pause_();
            }

          } catch (ex) {
            res = ex;
            this.hadError_ = true;

            if (!this.hasErrback_()) {
              // If an error is thrown with no additional errbacks in the queue,
              // prepare to rethrow the error.
              unhandledException = true;
            }
          }
        }
      }

      this.result_ = res;

      if (isChained && this.paused_) {
        res.addCallbacks(
            goog.bind(this.continue_, this, true /* isSuccess */),
            goog.bind(this.continue_, this, false /* isSuccess */));
        res.chained_ = true;
      }

      if (unhandledException) {
        // Rethrow the unhandled error after a timeout. Execution will continue, but
        // the error will be seen by global handlers and the user. The rethrow will
        // be canceled if another errback is appended before the timeout executes.
        this.unhandledExceptionTimeoutId_ = goog.global.setTimeout(function() {
          // The stack trace is clobbered when the error is rethrown. Append the
          // stack trace to the message if available. Since no one is capturing this
          // error, the stack trace will be printed to the debug console.
          if (goog.DEBUG && goog.isDef(res.message) && res.stack) {
            res.message += '\n' + res.stack;
          }
          throw res;
        }, 0);
      }
    };


    /**
     * Creates a deferred that always succeeds.
     * @param {*} res The result.
     * @return {!goog.async.Deferred} The deferred object.
     */
    goog.async.Deferred.succeed = function(res) {
      var d = new goog.async.Deferred();
      d.callback(res);
      return d;
    };


    /**
     * Creates a deferred that always fails.
     * @param {*} res The error result.
     * @return {!goog.async.Deferred} The deferred object.
     */
    goog.async.Deferred.fail = function(res) {
      var d = new goog.async.Deferred();
      d.errback(res);
      return d;
    };


    /**
     * An error sub class that is used when a deferred has already been called.
     * @param {!goog.async.Deferred} deferred The deferred object.
     * @constructor
     * @extends {goog.debug.Error}
     */
    goog.async.Deferred.AlreadyCalledError = function(deferred) {
      goog.debug.Error.call(this);

      /**
       * The deferred that raised this error.
       * @type {goog.async.Deferred}
       */
      this.deferred = deferred;
    };
    goog.inherits(goog.async.Deferred.AlreadyCalledError, goog.debug.Error);


    /**
     * Message text.
     * @type {string}
     * @override
     */
    goog.async.Deferred.AlreadyCalledError.prototype.message = 'Already called';



    /**
     * An error sub class that is used when a deferred is cancelled.
     * @param {!goog.async.Deferred} deferred The deferred object.
     * @constructor
     * @extends {goog.debug.Error}
     */
    goog.async.Deferred.CancelledError = function(deferred) {
      goog.debug.Error.call(this);

      /**
       * The deferred that raised this error.
       * @type {goog.async.Deferred}
       */
      this.deferred = deferred;
    };
    goog.inherits(goog.async.Deferred.CancelledError, goog.debug.Error);


    /**
     * Message text.
     * @type {string}
     * @override
     */
    goog.async.Deferred.CancelledError.prototype.message = 'Deferred was cancelled';
    "
    to contain:
      "goog.require('goog.Promise')" 
        at org_clojure.google_closure_library_third_party.Google_closure_library_third_partyTest.closureModulesKeepExpectedThirdPartyIntegrationPoints(Google_closure_library_third_partyTest.java:76)

Google_closure_library_third_partyTest > distributedResourceSetIsAvailableOnTheClasspath() FAILED
    java.lang.AssertionError: [resource goog/dojo/dom/query_test_dom.html should exist] 
    Expecting actual not to be null
        at org_clojure.google_closure_library_third_party.Google_closure_library_third_partyTest.readResource(Google_closure_library_third_partyTest.java:198)
        at org_clojure.google_closure_library_third_party.Google_closure_library_third_partyTest.distributedResourceSetIsAvailableOnTheClasspath(Google_closure_library_third_partyTest.java:52)

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/tests/src/org.clojure/google-closure-library-third-party/0.0-20250515-f04e4c0e/build/reports/problems/problems-report.html

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
13 actionable tasks: 3 executed, 10 up-to-date

> Task :javaTest FAILED

[Incubating] Problems report is available at: file:///home/runner/work/graalvm-reachability-metadata/graalvm-reachability-metadata/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':javaTest'.
> Java tests failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to generate a Build Scan (Powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.1.0/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
11 actionable tasks: 3 executed, 8 up-to-date
Finished stage 'java run' for 0.0-1376 at 2026-06-24T18:49:47Z with exit code 1 after 5s
❌ ERROR: 'java run' for 0.0-1376 failed with exit code 1.
FAILED[java run][0.0-1376][env GVM_TCK_LV=0.0-1376 GVM_TCK_NATIVE_IMAGE_MODE=future-defaults-all ./gradlew clean javaTest -Pcoordinates=org.clojure:google-closure-library-third-party:0.0-20250515-f04e4c0e]
```

--- a/metadata/org.clojure/google-closure-library-third-party/0.0-1376/reachability-metadata.json
+++ b/metadata/org.clojure/google-closure-library-third-party/0.0-1376/reachability-metadata.json
@@ -1,0 +1,16 @@
+{
+  "resources" : [
+    {
+      "glob" : "goog/**"
+    },
+    {
+      "glob" : "README"
+    },
+    {
+      "glob" : "AUTHORS"
+    },
+    {
+      "glob" : "LICENSE"
+    }
+  ]
+}

--- a/metadata/org.clojure/google-closure-library-third-party/index.json
+++ b/metadata/org.clojure/google-closure-library-third-party/index.json
@@ -1,6 +1,15 @@
 [
   {
     "latest" : true,
+    "metadata-version" : "0.0-1376",
+    "tested-versions" : [
+      "0.0-1376"
+    ],
+    "allowed-packages" : [
+      "org.clojure"
+    ]
+  },
+  {
     "metadata-version" : "0.0-20250515-f04e4c0e",
     "source-code-url" : "https://repo.maven.apache.org/maven2/org/clojure/google-closure-library-third-party/$version$/google-closure-library-third-party-$version$-sources.jar",
     "repository-url" : "https://github.com/google/closure-library",

--- a/metadata/org.clojure/google-closure-library-third-party/index.json
+++ b/metadata/org.clojure/google-closure-library-third-party/index.json
@@ -2,6 +2,11 @@
   {
     "latest" : true,
     "metadata-version" : "0.0-1376",
+    "source-code-url" : "https://repo.maven.apache.org/maven2/org/clojure/google-closure-library-third-party/$version$/google-closure-library-third-party-$version$.jar",
+    "repository-url" : "https://github.com/google/closure-library",
+    "test-code-url" : "https://repo.maven.apache.org/maven2/org/clojure/google-closure-library-third-party/$version$/google-closure-library-third-party-$version$.jar",
+    "documentation-url" : "https://github.com/google/closure-library/blob/14be8b286a11c5c85482fa88134cba765af78f5c/README",
+    "description" : "Google Closure Library Third-Party Extensions packages optional third-party JavaScript modules used with the Google Closure Library, including Dojo DOM query helpers, MochiKit async utilities, lorem ipsum data, Caja HTML handling, and Silverlight support files. This Maven artifact is a non-official ClojureScript-maintained distribution of those files for consumers that resolve Closure Library dependencies through Maven.",
     "tested-versions" : [
       "0.0-1376"
     ],

--- a/stats/org.clojure/google-closure-library-third-party/0.0-1376/execution-metrics.json
+++ b/stats/org.clojure/google-closure-library-third-party/0.0-1376/execution-metrics.json
@@ -1,0 +1,47 @@
+{
+  "fix_java_run_fail:2026-06-26": {
+    "timestamp": "2026-06-26T11:29:50.470146Z",
+    "library": "org.clojure:google-closure-library-third-party:0.0-1376",
+    "starting_commit": "d1622cff8fba16eace27845a70602b36e58f05cb",
+    "ending_commit": "447341a6cba91c60967115bdca86e764d07fd3a1",
+    "previous_library": "org.clojure:google-closure-library-third-party:0.0-20250515-f04e4c0e",
+    "strategy_name": "java_run_iterative_with_coverage_sources_pi_gpt-5.5",
+    "agent": "pi",
+    "model": "oca/gpt-5.5",
+    "status": "success",
+    "stats": {
+      "version": "0.0-1376",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      }
+    },
+    "previous_library_stats": {
+      "version": "0.0-20250515-f04e4c0e",
+      "dynamicAccess": {
+        "breakdown": {},
+        "coverageRatio": 1.0,
+        "coveredCalls": 0,
+        "totalCalls": 0
+      }
+    },
+    "metrics": {
+      "input_tokens_used": 88628,
+      "cached_input_tokens_used": 778240,
+      "output_tokens_used": 7005,
+      "iterations": 2,
+      "cost_usd": 0.5993,
+      "tested_library_loc": 0,
+      "metadata_entries": 4,
+      "previous_library_metadata_entries": 4,
+      "code_coverage_percent": 100.0,
+      "previous_library_coverage_percent": 100.0
+    },
+    "artifacts": {
+      "test_file": "tests/src/org.clojure/google-closure-library-third-party/0.0-1376/src/test/java/org_clojure/google_closure_library_third_party/Google_closure_library_third_partyTest.java",
+      "metadata_file": "metadata/org.clojure/google-closure-library-third-party/0.0-1376/reachability-metadata.json"
+    }
+  }
+}

--- a/stats/org.clojure/google-closure-library-third-party/0.0-1376/stats.json
+++ b/stats/org.clojure/google-closure-library-third-party/0.0-1376/stats.json
@@ -1,0 +1,16 @@
+{
+  "versions" : [ {
+    "version" : "0.0-1376",
+    "dynamicAccess" : {
+      "breakdown" : { },
+      "coverageRatio" : 1.0,
+      "coveredCalls" : 0,
+      "totalCalls" : 0
+    },
+    "libraryCoverage" : {
+      "instruction" : "N/A",
+      "line" : "N/A",
+      "method" : "N/A"
+    }
+  } ]
+}

--- a/tests/src/org.clojure/google-closure-library-third-party/0.0-1376/.gitignore
+++ b/tests/src/org.clojure/google-closure-library-third-party/0.0-1376/.gitignore
@@ -1,0 +1,4 @@
+gradlew.bat
+gradlew
+gradle/
+build/

--- a/tests/src/org.clojure/google-closure-library-third-party/0.0-1376/build.gradle
+++ b/tests/src/org.clojure/google-closure-library-third-party/0.0-1376/build.gradle
@@ -1,0 +1,27 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+plugins {
+    id "org.graalvm.internal.tck"
+}
+
+String libraryVersion = tck.testedLibraryVersion.get()
+
+dependencies {
+    testImplementation "org.clojure:google-closure-library-third-party:$libraryVersion"
+    testImplementation 'org.assertj:assertj-core:3.22.0'
+}
+
+graalvmNative {
+    agent {
+        defaultMode = "conditional"
+        modes {
+            conditional {
+                userCodeFilterPath = "user-code-filter.json"
+            }
+        }
+    }
+}

--- a/tests/src/org.clojure/google-closure-library-third-party/0.0-1376/gradle.properties
+++ b/tests/src/org.clojure/google-closure-library-third-party/0.0-1376/gradle.properties
@@ -1,0 +1,6 @@
+# Optional explicit tested library coordinates (format: group:artifact:version).
+# If omitted, fallback resolution in org.graalvm.internal.tck.gradle still uses
+# environment variable GVM_TCK_LC and then this property.
+library.coordinates = org.clojure:google-closure-library-third-party:0.0-1376
+library.version = 0.0-1376
+metadata.dir = org.clojure/google-closure-library-third-party/0.0-1376/

--- a/tests/src/org.clojure/google-closure-library-third-party/0.0-1376/settings.gradle
+++ b/tests/src/org.clojure/google-closure-library-third-party/0.0-1376/settings.gradle
@@ -1,0 +1,20 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+pluginManagement {
+    def tckPath = Objects.requireNonNullElse(
+            System.getenv("GVM_TCK_TCKDIR"),
+            "../../../../tck-build-logic"
+    )
+    includeBuild(tckPath)
+    repositories { mavenLocal(); gradlePluginPortal() }
+}
+
+plugins {
+    id "org.graalvm.internal.tck-settings" version "1.0.0-SNAPSHOT"
+}
+
+rootProject.name = 'org.clojure.google-closure-library-third-party_tests'

--- a/tests/src/org.clojure/google-closure-library-third-party/0.0-1376/src/test/java/org_clojure/google_closure_library_third_party/Google_closure_library_third_partyTest.java
+++ b/tests/src/org.clojure/google-closure-library-third-party/0.0-1376/src/test/java/org_clojure/google_closure_library_third_party/Google_closure_library_third_partyTest.java
@@ -1,0 +1,202 @@
+/*
+ * Copyright and related rights waived via CC0
+ *
+ * You should have received a copy of the CC0 legalcode along with this
+ * work. If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+ */
+package org_clojure.google_closure_library_third_party;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.entry;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.Test;
+
+public class Google_closure_library_third_partyTest {
+    private static final List<String> DISTRIBUTED_RESOURCES = List.of(
+            "goog/dojo/dom/query.js",
+            "goog/dojo/dom/query_test.js",
+            "goog/dojo/dom/query_test_dom.html",
+            "goog/mochikit/BUILD",
+            "goog/mochikit/LICENSE",
+            "goog/mochikit/async/BUILD",
+            "goog/mochikit/async/deferred.js",
+            "goog/mochikit/async/deferred_async_test.js",
+            "goog/mochikit/async/deferred_async_test_dom.html",
+            "goog/mochikit/async/deferred_test.js",
+            "goog/mochikit/async/deferredlist.js",
+            "goog/mochikit/async/deferredlist_test.js",
+            "README.md",
+            "AUTHORS",
+            "LICENSE");
+
+    private static final Map<String, List<String>> MODULE_PROVIDES = Map.of(
+            "goog/mochikit/async/deferred.js",
+            List.of(
+                    "goog.async.Deferred",
+                    "goog.async.Deferred.AlreadyCalledError",
+                    "goog.async.Deferred.CanceledError"),
+            "goog/mochikit/async/deferredlist.js",
+            List.of("goog.async.DeferredList"),
+            "goog/dojo/dom/query.js",
+            List.of("goog.dom.query"));
+
+    @Test
+    void distributedResourceSetIsAvailableOnTheClasspath() throws IOException {
+        for (String resourcePath : DISTRIBUTED_RESOURCES) {
+            String resource = readResource(resourcePath);
+
+            assertThat(resource)
+                    .as("resource %s should be packaged and readable", resourcePath)
+                    .isNotBlank();
+        }
+    }
+
+    @Test
+    void closureModulesDeclareExpectedPublicNamespaces() throws IOException {
+        for (Map.Entry<String, List<String>> module : MODULE_PROVIDES.entrySet()) {
+            String source = readResource(module.getKey());
+
+            for (String providedNamespace : module.getValue()) {
+                assertThat(source)
+                        .as("%s should provide %s", module.getKey(), providedNamespace)
+                        .contains("goog.provide('" + providedNamespace + "')");
+            }
+        }
+    }
+
+    @Test
+    void closureModulesKeepExpectedThirdPartyIntegrationPoints() throws IOException {
+        assertThat(readResource("goog/mochikit/async/deferred.js"))
+                .contains("goog.require('goog.Promise')")
+                .contains("goog.async.Deferred.prototype.callback")
+                .contains("goog.async.Deferred.prototype.errback")
+                .contains("goog.async.Deferred.prototype.cancel")
+                .contains("goog.async.Deferred.prototype.then")
+                .contains("goog.async.Deferred.fromPromise")
+                .contains("goog.async.Deferred.when");
+
+        assertThat(readResource("goog/mochikit/async/deferredlist.js"))
+                .contains("goog.require('goog.async.Deferred')")
+                .contains("goog.async.DeferredList.prototype.handleCallback_")
+                .contains("goog.async.DeferredList.prototype.errback")
+                .contains("goog.async.DeferredList.gatherResults");
+
+        assertThat(readResource("goog/dojo/dom/query.js"))
+                .contains("goog.dom.query = (function()")
+                .contains("This code was ported from the Dojo Toolkit")
+                .contains("querySelectorAll")
+                .contains("goog.exportSymbol('goog.dom.query', goog.dom.query)")
+                .contains("goog.exportSymbol('goog.dom.query.pseudos', goog.dom.query.pseudos)");
+    }
+
+    @Test
+    void deferredModuleCoversCancellationErrorsAndPromiseIntegration() throws IOException {
+        assertThat(readResource("goog/mochikit/async/deferred.js"))
+                .contains("goog.define('goog.async.Deferred.STRICT_ERRORS', false)")
+                .contains("goog.define('goog.async.Deferred.LONG_STACK_TRACES', false)")
+                .contains("goog.async.Deferred.prototype.makeStackTraceLong_")
+                .contains("goog.async.Deferred.AlreadyCalledError.prototype.name = 'AlreadyCalledError'")
+                .contains("goog.async.Deferred.CanceledError.prototype.name = 'CanceledError'")
+                .contains("goog.Thenable.addImplementation(goog.async.Deferred)")
+                .contains("goog.async.Deferred.assertNoErrors = function()");
+    }
+
+    @Test
+    void dojoQuerySelectorEngineSupportsCssSelectorsAttributesAndPseudos() throws IOException {
+        assertThat(readResource("goog/dojo/dom/query.js"))
+                .contains("'*=': function(attr, value)")
+                .contains("'^=': function(attr, value)")
+                .contains("'$=': function(attr, value)")
+                .contains("'~=': function(attr, value)")
+                .contains("'|=': function(attr, value)")
+                .contains("'=': function(attr, value)")
+                .contains("'first-child': function()")
+                .contains("'last-child': function()")
+                .contains("'only-child': function(name, condition)")
+                .contains("'empty': function(name, condition)")
+                .contains("'contains': function(name, condition)")
+                .contains("'not': function(name, condition)")
+                .contains("'nth-child': function(name, condition)");
+    }
+
+    @Test
+    void bundledJavascriptFixturesReferenceThePackagedModules() throws IOException {
+        assertThat(readResource("goog/mochikit/async/deferred_test.js"))
+                .contains("goog.module('goog.async.deferredTest')")
+                .contains("const Deferred = goog.require('goog.async.Deferred')")
+                .contains("testNormal()")
+                .contains("testCancel()")
+                .contains("testDeferredDependencies()");
+
+        assertThat(readResource("goog/mochikit/async/deferred_async_test.js"))
+                .contains("goog.module('goog.async.deferredAsyncTest')")
+                .contains("const Deferred = goog.require('goog.async.Deferred')")
+                .contains("testErrorStack()");
+
+        assertThat(readResource("goog/mochikit/async/deferredlist_test.js"))
+                .contains("goog.module('goog.async.deferredListTest')")
+                .contains("const DeferredList = goog.require('goog.async.DeferredList')")
+                .contains("testDeferredList()")
+                .contains("testGatherResults()");
+
+        assertThat(readResource("goog/mochikit/async/deferred_async_test_dom.html"))
+                .contains("goog.async.Deferred.LONG_STACK_TRACES = true;");
+
+        assertThat(readResource("goog/dojo/dom/query_test.js"))
+                .contains("function testBasicSelectors()")
+                .contains("function testNthChild()")
+                .contains("function testAttributes()");
+
+        assertThat(readResource("goog/dojo/dom/query_test_dom.html"))
+                .contains("<h1>testing goog.dom.query()</h1>")
+                .contains("id=iframe-test");
+    }
+
+    @Test
+    void projectDocumentationAndLicenseArePackaged() throws IOException {
+        assertThat(readResource("README.md"))
+                .contains("# Closure Library")
+                .contains("*actively* maintained by the Clojure team")
+                .contains("Google stopped contributing to Closure Library on August 2024")
+                .contains("Previous version of this README can be found here");
+
+        assertThat(readResource("AUTHORS"))
+                .contains("Closure Library")
+                .contains("Google Inc.");
+
+        assertThat(readResource("LICENSE"))
+                .contains("Apache License")
+                .contains("Version 2.0, January 2004")
+                .contains("TERMS AND CONDITIONS FOR USE, REPRODUCTION, AND DISTRIBUTION");
+    }
+
+    @Test
+    void packagedModulesRemainInExpectedThirdPartyGroups() {
+        Map<String, Long> resourceCountsByGroup = DISTRIBUTED_RESOURCES.stream()
+                .filter(resourcePath -> resourcePath.startsWith("goog/"))
+                .collect(Collectors.groupingBy(
+                        resourcePath -> resourcePath.substring(0, resourcePath.indexOf('/', "goog/".length())),
+                        Collectors.counting()));
+
+        assertThat(resourceCountsByGroup)
+                .contains(
+                        entry("goog/dojo", 3L),
+                        entry("goog/mochikit", 9L));
+    }
+
+    private static String readResource(String resourcePath) throws IOException {
+        ClassLoader classLoader = Google_closure_library_third_partyTest.class.getClassLoader();
+        try (InputStream resourceStream = classLoader.getResourceAsStream(resourcePath)) {
+            assertThat(resourceStream)
+                    .as("resource %s should exist", resourcePath)
+                    .isNotNull();
+            return new String(resourceStream.readAllBytes(), StandardCharsets.UTF_8);
+        }
+    }
+}

--- a/tests/src/org.clojure/google-closure-library-third-party/0.0-1376/src/test/java/org_clojure/google_closure_library_third_party/Google_closure_library_third_partyTest.java
+++ b/tests/src/org.clojure/google-closure-library-third-party/0.0-1376/src/test/java/org_clojure/google_closure_library_third_party/Google_closure_library_third_partyTest.java
@@ -19,19 +19,28 @@ import org.junit.jupiter.api.Test;
 
 public class Google_closure_library_third_partyTest {
     private static final List<String> DISTRIBUTED_RESOURCES = List.of(
+            "goog/base.js",
+            "goog/deps.js",
+            "goog/caja/string/html/htmlparser.js",
+            "goog/caja/string/html/htmlsanitizer.js",
             "goog/dojo/dom/query.js",
             "goog/dojo/dom/query_test.js",
-            "goog/dojo/dom/query_test_dom.html",
-            "goog/mochikit/BUILD",
-            "goog/mochikit/LICENSE",
-            "goog/mochikit/async/BUILD",
+            "goog/dojo/dom/query_test.html",
+            "goog/jpeg_encoder/jpeg_encoder_basic.js",
+            "goog/loremipsum/text/loremipsum.js",
+            "goog/loremipsum/text/loremipsum_test.html",
             "goog/mochikit/async/deferred.js",
-            "goog/mochikit/async/deferred_async_test.js",
-            "goog/mochikit/async/deferred_async_test_dom.html",
-            "goog/mochikit/async/deferred_test.js",
+            "goog/mochikit/async/deferred_test.html",
             "goog/mochikit/async/deferredlist.js",
-            "goog/mochikit/async/deferredlist_test.js",
-            "README.md",
+            "goog/mochikit/async/deferredlist_test.html",
+            "goog/osapi/osapi.js",
+            "goog/silverlight/AppManifest.xml",
+            "goog/silverlight/ClipboardButton.xaml",
+            "goog/silverlight/ClipboardButtonApp.xaml",
+            "goog/silverlight/clipboardbutton.js",
+            "goog/silverlight/silverlight.js",
+            "goog/silverlight/supporteduseragent.js",
+            "README",
             "AUTHORS",
             "LICENSE");
 
@@ -40,7 +49,7 @@ public class Google_closure_library_third_partyTest {
             List.of(
                     "goog.async.Deferred",
                     "goog.async.Deferred.AlreadyCalledError",
-                    "goog.async.Deferred.CanceledError"),
+                    "goog.async.Deferred.CancelledError"),
             "goog/mochikit/async/deferredlist.js",
             List.of("goog.async.DeferredList"),
             "goog/dojo/dom/query.js",
@@ -73,13 +82,14 @@ public class Google_closure_library_third_partyTest {
     @Test
     void closureModulesKeepExpectedThirdPartyIntegrationPoints() throws IOException {
         assertThat(readResource("goog/mochikit/async/deferred.js"))
-                .contains("goog.require('goog.Promise')")
                 .contains("goog.async.Deferred.prototype.callback")
                 .contains("goog.async.Deferred.prototype.errback")
                 .contains("goog.async.Deferred.prototype.cancel")
-                .contains("goog.async.Deferred.prototype.then")
-                .contains("goog.async.Deferred.fromPromise")
-                .contains("goog.async.Deferred.when");
+                .contains("goog.async.Deferred.prototype.chainDeferred")
+                .contains("goog.async.Deferred.prototype.awaitDeferred")
+                .contains("goog.async.Deferred.prototype.branch")
+                .contains("goog.async.Deferred.succeed")
+                .contains("goog.async.Deferred.fail");
 
         assertThat(readResource("goog/mochikit/async/deferredlist.js"))
                 .contains("goog.require('goog.async.Deferred')")
@@ -96,15 +106,17 @@ public class Google_closure_library_third_partyTest {
     }
 
     @Test
-    void deferredModuleCoversCancellationErrorsAndPromiseIntegration() throws IOException {
+    void deferredModuleCoversCancellationErrorsAndCallbackChaining() throws IOException {
         assertThat(readResource("goog/mochikit/async/deferred.js"))
-                .contains("goog.define('goog.async.Deferred.STRICT_ERRORS', false)")
-                .contains("goog.define('goog.async.Deferred.LONG_STACK_TRACES', false)")
-                .contains("goog.async.Deferred.prototype.makeStackTraceLong_")
-                .contains("goog.async.Deferred.AlreadyCalledError.prototype.name = 'AlreadyCalledError'")
-                .contains("goog.async.Deferred.CanceledError.prototype.name = 'CanceledError'")
-                .contains("goog.Thenable.addImplementation(goog.async.Deferred)")
-                .contains("goog.async.Deferred.assertNoErrors = function()");
+                .contains("this.canceller_.call(this.defaultScope_, this)")
+                .contains("this.errback(new goog.async.Deferred.CancelledError(this))")
+                .contains("throw new goog.async.Deferred.AlreadyCalledError(this)")
+                .contains("goog.async.Deferred.prototype.addCallbacks")
+                .contains("goog.async.Deferred.prototype.addErrback")
+                .contains("goog.async.Deferred.prototype.addBoth")
+                .contains("goog.async.Deferred.AlreadyCalledError.prototype.message = 'Already called'")
+                .contains("goog.async.Deferred.CancelledError.prototype.message = 'Deferred was cancelled'")
+                .contains("goog.global.setTimeout(function()");
     }
 
     @Test
@@ -127,44 +139,35 @@ public class Google_closure_library_third_partyTest {
 
     @Test
     void bundledJavascriptFixturesReferenceThePackagedModules() throws IOException {
-        assertThat(readResource("goog/mochikit/async/deferred_test.js"))
-                .contains("goog.module('goog.async.deferredTest')")
-                .contains("const Deferred = goog.require('goog.async.Deferred')")
-                .contains("testNormal()")
-                .contains("testCancel()")
-                .contains("testDeferredDependencies()");
+        assertThat(readResource("goog/mochikit/async/deferred_test.html"))
+                .contains("goog.require('goog.async.Deferred')")
+                .contains("var Deferred = goog.async.Deferred")
+                .contains("function testNormal()")
+                .contains("function testCancel()")
+                .contains("function testDeferredDependencies()");
 
-        assertThat(readResource("goog/mochikit/async/deferred_async_test.js"))
-                .contains("goog.module('goog.async.deferredAsyncTest')")
-                .contains("const Deferred = goog.require('goog.async.Deferred')")
-                .contains("testErrorStack()");
-
-        assertThat(readResource("goog/mochikit/async/deferredlist_test.js"))
-                .contains("goog.module('goog.async.deferredListTest')")
-                .contains("const DeferredList = goog.require('goog.async.DeferredList')")
-                .contains("testDeferredList()")
-                .contains("testGatherResults()");
-
-        assertThat(readResource("goog/mochikit/async/deferred_async_test_dom.html"))
-                .contains("goog.async.Deferred.LONG_STACK_TRACES = true;");
+        assertThat(readResource("goog/mochikit/async/deferredlist_test.html"))
+                .contains("goog.require('goog.async.DeferredList')")
+                .contains("var DeferredList = goog.async.DeferredList")
+                .contains("function testDeferredList()")
+                .contains("function testGatherResults()");
 
         assertThat(readResource("goog/dojo/dom/query_test.js"))
                 .contains("function testBasicSelectors()")
                 .contains("function testNthChild()")
                 .contains("function testAttributes()");
 
-        assertThat(readResource("goog/dojo/dom/query_test_dom.html"))
+        assertThat(readResource("goog/dojo/dom/query_test.html"))
                 .contains("<h1>testing goog.dom.query()</h1>")
                 .contains("id=iframe-test");
     }
 
     @Test
     void projectDocumentationAndLicenseArePackaged() throws IOException {
-        assertThat(readResource("README.md"))
-                .contains("# Closure Library")
-                .contains("*actively* maintained by the Clojure team")
-                .contains("Google stopped contributing to Closure Library on August 2024")
-                .contains("Previous version of this README can be found here");
+        assertThat(readResource("README"))
+                .contains("Closure Library is a powerful, low level JavaScript library designed")
+                .contains("for building complex and scalable web applications")
+                .contains("http://code.google.com/closure/library");
 
         assertThat(readResource("AUTHORS"))
                 .contains("Closure Library")
@@ -180,14 +183,20 @@ public class Google_closure_library_third_partyTest {
     void packagedModulesRemainInExpectedThirdPartyGroups() {
         Map<String, Long> resourceCountsByGroup = DISTRIBUTED_RESOURCES.stream()
                 .filter(resourcePath -> resourcePath.startsWith("goog/"))
+                .filter(resourcePath -> resourcePath.indexOf('/', "goog/".length()) > 0)
                 .collect(Collectors.groupingBy(
                         resourcePath -> resourcePath.substring(0, resourcePath.indexOf('/', "goog/".length())),
                         Collectors.counting()));
 
         assertThat(resourceCountsByGroup)
                 .contains(
+                        entry("goog/caja", 2L),
                         entry("goog/dojo", 3L),
-                        entry("goog/mochikit", 9L));
+                        entry("goog/jpeg_encoder", 1L),
+                        entry("goog/loremipsum", 2L),
+                        entry("goog/mochikit", 4L),
+                        entry("goog/osapi", 1L),
+                        entry("goog/silverlight", 6L));
     }
 
     private static String readResource(String resourcePath) throws IOException {

--- a/tests/src/org.clojure/google-closure-library-third-party/0.0-1376/user-code-filter.json
+++ b/tests/src/org.clojure/google-closure-library-third-party/0.0-1376/user-code-filter.json
@@ -1,0 +1,10 @@
+{
+  "rules" : [
+    {
+      "excludeClasses" : "**"
+    },
+    {
+      "includeClasses" : "org.clojure.**"
+    }
+  ]
+}

--- a/tests/src/org.clojure/google-closure-library-third-party/0.0-1376/user-code-filter.json
+++ b/tests/src/org.clojure/google-closure-library-third-party/0.0-1376/user-code-filter.json
@@ -5,6 +5,9 @@
     },
     {
       "includeClasses" : "org.clojure.**"
+    },
+    {
+      "includeClasses" : "org_clojure.google_closure_library_third_party.**"
     }
   ]
 }

